### DISCUSSION TITLE
Git Repo Access issue with application set created through an application which has repo access

### DISCUSSION THREAD

**Discussion by rmehta2974** [argoproj/argo-cd#18644] [General]
apiVersion: argoproj.io/v1alpha1
kind: Application
metadata:
name: certmanager-deploy-application
namespace: argocd
spec:
project: default
source:
repoURL: [git@github.com](mailto:git@github.com)/xxx/xxx-xxxx-v2.git
targetRevision: az-528-1
path: cluster/deployment/non-prod/apps/
directory:
recurse: true
destination:
server: https://kubernetes.default.svc/
namespace: argocd
syncPolicy: {}

which reads the repo and configuration correctly

but I have application set in the search path cluster/deployment/non-prod/apps/cert-manager/cert-manager.yaml

apiVersion: argoproj.io/v1alpha1
kind: ApplicationSet
metadata:
name: cert-manager-multicluster-list2
namespace: argocd
spec:
generators:
- matrix:
generators:

      - git:
          repoURL: git@github.com/xxx/xxx-xxxx-v2.git
          revision: az-528-1
          directories:
            - path: cluster/deployment/non-prod/apps/cert-manager/*

      - list:
         elements:
          - cluster: dev0-ccp-aks
            namespace: cert-manager
            server: https://xxxx.xxxx.xxx
            targetRevision: v1.13.2
            valuesFile: dev0-ccp-aks-values.yaml
          - cluster: management
            namespace: cert-manager
            server: https://kubernetes.default.svc
            targetRevision: v1.13.2
            valuesFile: management-values.yaml
template:
metadata:
name: cert-manager-{{cluster}}
spec:
project: default
source:
repoURL: https://charts.jetstack.io/
chart: cert-manager
targetRevision: "{{targetRevision}}"
helm:
valueFiles:
- cluster/argocd/non-prod/apps/cert-manager/common-values.yaml
- cluster/argocd/non-prod/apps/cert-manager/{{valuesFile}}
destination:
namespace: "{{namespace}}"
server: "{{server}}"
syncPolicy:
automated:
prune: true
selfHeal: true
syncOptions:
- CreateNamespace=true

Now above code gives following error

error failed to get params for first generator in matrix generator: child generator returned an error on parameter generation: error generating params from git: error getting directories from repo: error retrieving Git Directories: rpc error: code = Internal desc = unable to resolve git revision : repository not found

Here is output of argocd repo list

**Comment by dustinvanbuskirk** [comment]
I ran into this issue when trying to setup an ApplicationSet file in one repository designated for ArgoCD Applications and then sourced the git generator from folders located in a second git repository meant for configurations in my case folders with Helm Charts and Values files.

I had to add a repo https://github.com/<org> as a credentials template and this got me past this error.  I have both the applications and configurations repositories added as repositories with successful connections but only after adding the org as a credentials template was the git generator able to authenticate.

Hope this is helpful for others.
